test/built-ins/String/prototype/matchAll/regexp-prototype-matchAll-v-flag.js

```diff
@@ -2,42 +2,82 @@
 // This code is governed by the BSD license found in the LICENSE file.
 /*---
 esid: sec-regexp.prototype-@@matchall
-description: RegExp.prototype[@@matchAll] behavior with 'v' flag
-features: [Symbol.matchAll, regexp-v-flag]
+description: RegExp.prototype[@@matchAll] behavior with 'u'and 'v' flags.
+features: [Symbol.matchAll, regexp-v-flag, regexp-unicode-property-escapes]
 ---*/
 
 const text = '𠮷a𠮷b𠮷';
 
 function doMatchAll(regex) {
-  return Array.from(RegExp.prototype[Symbol.matchAll].call(regex, text), m => [m[0], m.index]);
+  const result = Array.from(RegExp.prototype[Symbol.matchAll].call(regex, text));
+  const matches = result.map(m => m[0]);
+  const indices = result.map(m => m.index);
+  return matches.concat(indices);
 }
 
 assert.sameValue(
-  doMatchAll(/𠮷/g).toString(),
-  "𠮷,0,𠮷,3,𠮷,6",
+  assert.compareArray(doMatchAll(/𠮷/g), ['𠮷', '𠮷', '𠮷', 0, 3, 6]),
+  undefined,
   "Basic matchAll with g flag"
 );
 
 assert.sameValue(
-  doMatchAll(/𠮷/gv).toString(),
-  "𠮷,0,𠮷,3,𠮷,6",
+  assert.compareArray(doMatchAll(/𠮷/gu), ['𠮷', '𠮷', '𠮷', 0, 3, 6]),
+  undefined,
+  "matchAll with u flag"
+);
+
+assert.sameValue(
+  assert.compareArray(doMatchAll(/𠮷/gv), ['𠮷', '𠮷', '𠮷', 0, 3, 6]),
+  undefined,
   "matchAll with v flag"
 );
 
 assert.sameValue(
-  doMatchAll(/\p{Script=Han}/gv).toString(),
-  "𠮷,0,𠮷,3,𠮷,6",
+  assert.compareArray(doMatchAll(/\p{Script=Han}/gu), ['𠮷', '𠮷', '𠮷', 0, 3, 6]),
+  undefined,
+  "Unicode property escapes with u flag"
+);
+
+assert.sameValue(
+  assert.compareArray(doMatchAll(/\p{Script=Han}/gv), ['𠮷', '𠮷', '𠮷', 0, 3, 6]),
+  undefined,
   "Unicode property escapes with v flag"
 );
 
 assert.sameValue(
-  doMatchAll(/./gv).toString(),
-  "𠮷,0,a,2,𠮷,3,b,5,𠮷,6",
+  assert.compareArray(doMatchAll(/./gu), ['𠮷', 'a', '𠮷', 'b', '𠮷', 0, 2, 3, 5, 6]),
+  undefined,
+  "Dot with u flag"
+);
+
+assert.sameValue(
+  assert.compareArray(doMatchAll(/./gv), ['𠮷', 'a', '𠮷', 'b', '𠮷', 0, 2, 3, 5, 6]),
+  undefined,
   "Dot with v flag"
 );
 
+assert.sameValue(
+  doMatchAll(/(?:)/gu).length,
+  6,
+  "Empty matches with u flag"
+);
+
 assert.sameValue(
   doMatchAll(/(?:)/gv).length,
   6,
   "Empty matches with v flag"
 );
+
+const complexText = 'a\u{20BB7}b\u{10FFFF}c';
+assert.sameValue(
+  assert.compareArray(Array.from(complexText.matchAll(/\P{ASCII}/gu), m => m[0]), ['\u{20BB7}', '\u{10FFFF}']),
+  undefined,
+  "Non-ASCII with u flag"
+);
+assert.sameValue(
+  assert.compareArray(Array.from(complexText.matchAll(/\P{ASCII}/gv), m => m[0]), ['\u{20BB7}', '\u{10FFFF}']),
+  undefined,
+  "Non-ASCII with v flag"
+);
+reportCompare(0, 0);
```

Comment:
```suggestion
reportCompare(0, 0);

```

The lint test seems to be failing due to this missing newline in the end of the file.

Reply:
Thanks for the guidance.